File: tasks/builtin/refactor-01-go/starter.go
package solution

type Record map[string]interface{}

type Result struct {
	ActiveTotal               float64
	HighPriorityTotal         float64
	PendingTotal              float64
	ActiveHighPriorityCount   int
	PendingHighPriorityCount  int
}

func ProcessRecords(records []Record) Result {
	activeTotal := 0.0
	for _, record := range records {
		if status, ok := record["status"].(string); ok && status == "active" {
			if amount, ok := record["amount"].(float64); ok && amount > 0 {
				activeTotal += amount
			}
		}
	}

	highPriorityTotal := 0.0
	for _, record := range records {
		if priority, ok := record["priority"].(string); ok && priority == "high" {
			if amount, ok := record["amount"].(float64); ok && amount > 0 {
				highPriorityTotal += amount
			}
		}
	}

	pendingTotal := 0.0
	for _, record := range records {
		if status, ok := record["status"].(string); ok && status == "pending" {
			if amount, ok := record["amount"].(float64); ok && amount > 0 {
				pendingTotal += amount
			}
		}
	}

	activeHighPriorityCount := 0
	for _, record := range records {
		if status, ok := record["status"].(string); ok && status == "active" {
			if priority, ok := record["priority"].(string); ok && priority == "high" {
				activeHighPriorityCount++
			}
		}
	}

	pendingHighPriorityCount := 0
	for _, record := range records {
		if status, ok := record["status"].(string); ok && status == "pending" {
			if priority, ok := record["priority"].(string); ok && priority == "high" {
				pendingHighPriorityCount++
			}
		}
	}

	return Result{
		ActiveTotal:              activeTotal,
		HighPriorityTotal:        highPriorityTotal,
		PendingTotal:             pendingTotal,
		ActiveHighPriorityCount:  activeHighPriorityCount,
		PendingHighPriorityCount: pendingHighPriorityCount,
	}
}

File: tasks/builtin/refactor-01-js/starter.js
function processRecords(records) {
    let activeTotal = 0;
    for (const record of records) {
        if (record.status === 'active' && record.amount !== undefined) {
            const amount = record.amount;
            if (amount > 0) {
                activeTotal += amount;
            }
        }
    }

    let highPriorityTotal = 0;
    for (const record of records) {
        if (record.priority === 'high' && record.amount !== undefined) {
            const amount = record.amount;
            if (amount > 0) {
                highPriorityTotal += amount;
            }
        }
    }

    let pendingTotal = 0;
    for (const record of records) {
        if (record.status === 'pending' && record.amount !== undefined) {
            const amount = record.amount;
            if (amount > 0) {
                pendingTotal += amount;
            }
        }
    }

    let activeHighPriorityCount = 0;
    for (const record of records) {
        if (record.status === 'active' && record.priority === 'high') {
            activeHighPriorityCount++;
        }
    }

    let pendingHighPriorityCount = 0;
    for (const record of records) {
        if (record.status === 'pending' && record.priority === 'high') {
            pendingHighPriorityCount++;
        }
    }

    return {
        activeTotal,
        highPriorityTotal,
        pendingTotal,
        activeHighPriorityCount,
        pendingHighPriorityCount
    };
}

module.exports = { processRecords };
